msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/JsonHelper.java

```diff
@@ -40,15 +41,77 @@ static <T> T convertJsonToObject(final String json, final Class<T> tClass) {
     }
 
     static IdToken createIdTokenFromEncodedTokenString(String token) {
-        String idTokenJson;
+        return JsonHelper.convertJsonToObject(getTokenPayloadClaims(token), IdToken.class);
```

Comment:
Would a log help here in a try catch block to clarify that parsing id token failed?